src/cpp/client/client_context.cc

```diff
@@ -178,4 +180,10 @@ void ClientContext::SetGlobalCallbacks(GlobalCallbacks* client_callbacks) {
   g_client_callbacks = client_callbacks;
 }
 
+bool ClientContext::trailers_only() const {
+  return initial_metadata_received_ &&
+         recv_initial_metadata_.arr()->count == 0 &&
```

Comment:
Is there actually any need to check this?  It seems like this will always inherently be true if `grpc_call_is_trailers_only()` returns true.

Reply:
You're obviously right. My mental model was that for a 0 message, that init metadata and trailing metadata would be pressed together into the trailers, but that would obviously make any init metadata into trailing metadata. It's almost like we should have a DEBUG_ASSERT in place here to specify that if this is true that the array count had better be 0.